src/main/kotlin/snyk/container/annotator/ContainerYamlAnnotator.kt

```diff
@@ -0,0 +1,92 @@
+package snyk.container.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.ExternalAnnotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+import io.snyk.plugin.getSnykToolWindowPanel
+import org.jetbrains.annotations.TestOnly
+import snyk.container.ContainerIssuesForImage
+
+class ContainerYamlAnnotator : ExternalAnnotator<PsiFile, Unit>() {
+    private val logger = logger<ContainerYamlAnnotator>()
+
+    // overrides needed for the Annotator to invoke apply(). We don't do anything here
+    override fun collectInformation(file: PsiFile): PsiFile? = file
+    override fun doAnnotate(collectedInfo: PsiFile?): Unit = Unit
+
+    fun getContainerIssuesForImages(psiFile: PsiFile): List<ContainerIssuesForImage> {
+        logger.debug("Calling doAnnotate on ${psiFile.name}")
+
+        val containerResult = getSnykToolWindowPanel(psiFile.project)?.currentContainerResult
+
+        ProgressManager.checkCanceled()
+
+        return containerResult?.allCliIssues
+            ?.filter { forImage -> forImage.workloadImages.find { it.psiFile == psiFile } != null }
+            ?: emptyList()
+    }
+
+    override fun apply(psiFile: PsiFile, annotationResult: Unit, holder: AnnotationHolder) {
+        val issuesForImages = getContainerIssuesForImages(psiFile)
+
+        if (issuesForImages.isEmpty()) return
+
+        issuesForImages.forEach { forImage ->
+            if (forImage.ignored || forImage.obsolete) return
+
+            val severity = severity(forImage)
+
+            val workloadImage = forImage.workloadImages.first { it.psiFile == psiFile }
+            holder.newAnnotation(severity, annotationMessage(forImage))
+                .range(textRange(psiFile, workloadImage.lineNumber, forImage.imageName))
+                .create()
+        }
+    }
+
+    @TestOnly
+    fun severity(forImage: ContainerIssuesForImage): HighlightSeverity {
+        val severities = forImage.vulnerabilities.groupBy { it.severity }
+        return when {
+            severities[SEVERITY_CRITICAL]?.isNotEmpty() == true -> HighlightSeverity.ERROR
+            severities[SEVERITY_HIGH]?.isNotEmpty() == true -> HighlightSeverity.WARNING
+            severities[SEVERITY_MEDIUM]?.isNotEmpty() == true -> HighlightSeverity.WEAK_WARNING
+            severities[SEVERITY_LOW]?.isNotEmpty() == true -> HighlightSeverity.INFORMATION
+            else -> HighlightSeverity.INFORMATION
+        }
+    }
+
+    fun annotationMessage(image: ContainerIssuesForImage): String {
```

Comment:
internal ?

Reply:
needed for testing - not possible :)

Reply:
hmm.. test is in the same package, so `internal` should works for both class and test usage.

Reply:
Tried it out - compile error

Reply:
(and thought exactly the same - I changed it first, but didn't work)

Reply:
ah, it's cause it in integTest that doesn't see as the same package for some reason. Ok. let's leave it public then